File: 1twosum/twosum.cpp
#include <map>
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> sum_map;
        vector<int> a(2);
        for(int i = 0,one_num = 0; i < nums.size(); i++){
            one_num = nums[i];
            map<int,int>::iterator iter;
            if(((iter = sum_map.find(target - one_num)) != sum_map.end())){
                a[0] = iter->second;
                a[1] = i;
                return a;
            } else{
                sum_map[one_num] = i; 
            }
        }
        return a;
    }
};

File: 1twosum/twosum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<Integer, Integer>();
        int find;
        for( int i = 0; i < nums.length; i++) {
            find = target - nums[i];
            if(map.containsKey(find)) {
                return new int[]{map.get(find), i};
            }
            map.put(nums[i], i);
        }
        return null;
    }
}